Refactor Sqlalchemy queries to 2.0 style (Part 5)

phanikumv opened the pull request:
This is a continuation of the effort to refactor the queries to sqlalchemy 2.0

related: https://github.com/apache/airflow/issues/28723

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

phanikumv committed:
SqlAlchemy 2.0 changes

phanikumv committed:
changes to jobs command

phanikumv committed:
Change task command

phanikumv committed:
Changes to variable command

phanikumv committed:
Change fernet key command

uranusjr committed:
Some minor SQL query fixups

uranusjr reviewed (approved):
I made some minor changes.

phanikumv committed:
Fix tests

uranusjr committed:
Work around ORM mocking

phanikumv committed:
More changes to renderedtifields

phanikumv committed:
Fix tests

uranusjr commented on tests/cli/commands/test_task_command.py:
```diff
@@ -500,15 +500,14 @@ def test_mapped_task_render_with_template(self, dag_maker):
         assert 'echo "2022-01-08"' in output
 
-    @mock.patch("sqlalchemy.orm.session.Session.query")
+    @mock.patch("airflow.cli.commands.task_command.select")
```
I don’t think you should do this; in practice `select` never raises `NoResultsFound` when it is called, only when its result is submitted for execution (via `execute`, `scalars`, etc.).

phanikumv replied:
Replaced it with `` @mock.patch("airflow.cli.commands.task_command.Session.scalars")``

phanikumv committed:
Apply review suggestions

phanikumv committed:
Apply review suggestions

uranusjr commented on airflow/models/renderedtifields.py:
```diff
@@ -236,14 +232,15 @@ def _do_delete_old_records(
                 tuple_not_in_condition(
                     (cls.dag_id, cls.task_id, cls.run_id),
-                    session.query(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
+                    select(ti_clause.c.dag_id, ti_clause.c.task_id, ti_clause.c.run_id),
```
Damn, `tuple_not_in_condition` and `tuple_in_condition` do not work well with the new style. Let’s leave this out, I’ll try to convert them in a separate PR.

phanikumv replied:
it's failing specifically for MSSQL, for rest all databases it is working fine. Will check today if we need to have specific logic for that

uranusjr replied:
I figured this out, will push to this branch

uranusjr commented on airflow/models/renderedtifields.py:
```diff
@@ -236,14 +232,15 @@ def _do_delete_old_records(
         session: Session,
     ) -> None:
         # This query might deadlock occasionally and it should be retried if fails (see decorator)
+        # breakpoint()
```
Accident?

phanikumv replied:
Removed it

uranusjr committed:
Convert tuple_in_condition to SQLAlchemy 2

Also tuple_not_in_condition since they are basically the same thing but
different operator.

phanikumv committed:
Remove commented code
